compiler/semmagic.nim

```diff
@@ -535,7 +535,28 @@ proc magicsAfterOverloadResolution(c: PContext, n: PNode,
             getAttachedOp(c.graph, t, attachedDestructor).owner == fin:
           discard "already turned this one into a finalizer"
         else:
-          bindTypeHook(c, turnFinalizerIntoDestructor(c, fin, n.info), n, attachedDestructor)
+          if sfForward in fin.flags:
```

Comment:
Move the full `of mNewFinalize` logic into its own proc please.

Reply:
done